src/software/ai/passing/pass_generator.cpp

```diff
@@ -0,0 +1,179 @@
+        for (const auto& [robot_id, receiving_positions] : receiving_positions_map)
+        {
+            for (const Point& receiving_position : receiving_positions)
+            {
+                debug_shapes.push_back(*createDebugShape(
+                    Stadium(world.friendlyTeam().getRobotById(robot_id)->position(),
+                            receiving_position, 0.02),
+                    std::to_string(debug_shapes.size()) + "pg"));
+            }
+        }
+        std::stringstream stream;
+        stream << "BP:" << std::fixed << std::setprecision(3) << best_pass.rating;
+        debug_shapes.push_back(
+            *createDebugShape(Circle(best_pass.pass.receiverPoint(), 0.05),
+                              std::to_string(debug_shapes.size()) + "pg", stream.str()));
+        LOG(VISUALIZE) << *createDebugShapes(debug_shapes);
+    }
+
+    // Generate sample passes across the field for cost visualization
+    if (passing_config_.cost_vis_config().generate_sample_passes())
+    {
+        samplePassesForVisualization(world, passing_config_, best_pass.pass);
+    }
+
+    return best_pass;
+}
+
+std::map<RobotId, std::vector<Point>> PassGenerator::sampleReceivingPositionsPerRobot(
+    const World& world, const std::vector<RobotId>& robots_to_ignore)
+{
+    std::map<RobotId, std::vector<Point>> receiving_positions_map;
+
+    const double min_sampling_std_dev =
+        passing_config_.pass_gen_min_rand_sample_std_dev_meters();
+    const double sampling_std_dev_vel_multiplier =
+        passing_config_.pass_gen_rand_sample_std_dev_robot_vel_multiplier();
+    const double sampling_center_vel_multiplier =
+        passing_config_.pass_gen_rand_sample_center_robot_vel_multiplier();
+
+    for (const Robot& robot : world.friendlyTeam().getAllRobots())
+    {
+        // Ignore robots in the ignore list
+        if (std::find(robots_to_ignore.begin(), robots_to_ignore.end(), robot.id()) !=
+            robots_to_ignore.end())
+        {
+            continue;
+        }
+
+        // Add the robot's current position to the list of sampled passes
+        auto robot_position = robot.position();
+        receiving_positions_map.insert({robot.id(), {robot_position}});
+
+        // Add the best pass from the previous iteration to the list of sampled passes
+        if (previous_best_receiving_positions_.find(robot.id()) !=
+            previous_best_receiving_positions_.end())
+        {
+            receiving_positions_map[robot.id()].push_back(
+                previous_best_receiving_positions_[robot.id()]);
+        }
+
+        // Sample random receiving positions using a normal distribution around the
+        // robot's future position with a standard deviation that scales with the robot's
+        // velocity.
+        Point sampling_center =
+            robot_position + (robot.velocity() * sampling_center_vel_multiplier);
+        double std_dev = min_sampling_std_dev +
```

Comment:
Also, I feel like this would be a `const`?

Reply:
If this is a const, I think we should change the name and move it to line 73?

Reply:
I've updated it to be const, and I thought it would probably make most sense for the normal distribution mean and std-dev to be initialized next to eachother for clarity.